src/core/thread/mesh_forwarder.cpp

```diff
@@ -894,6 +932,61 @@ otError MeshForwarder::SendEmptyFrame(Mac::Frame &aFrame, bool aAckRequest)
     return OT_ERROR_NONE;
 }
 
+otError MeshForwarder::GetFramePriority(uint8_t *               aFrame,
+                                        uint8_t                 aFrameLength,
+                                        const Mac::Address &    aMacSource,
+                                        const Mac::Address &    aMacDest,
+                                        const otThreadLinkInfo &aLinkInfo,
+                                        uint8_t &               aPriority)
+{
+    otError error = OT_ERROR_NONE;
+#if OPENTHREAD_ENABLE_QOS
+    Message *              message;
+    Lowpan::MeshHeader     meshHeader;
+    Lowpan::FragmentHeader fragmentHeader;
+    Ip6::Header            ip6Header;
+
+    // Skip mesh header
+    if (aFrameLength >= 1 && reinterpret_cast<Lowpan::MeshHeader *>(aFrame)->IsMeshHeader())
```

Comment:
This duplicates a lot of code with what is in `MeshForwarder::GetIp6Header()`.  Can we reduce code duplication?

Reply:
I will try to optimize the code.